### ISSUE TITLE
Add the ability to attach files

### ISSUE THREAD

**Issue by bserrao** [athensresearch/athens#878]
I would like to have the ability to attach files: images, documents, videos, ...
Those files would be "linked" to the block they're added and stored locally (where athens is running). It could be some drag and drop, an upload file button or shortcut (/file) which launch a local files find or browse your computer option, ...

Additionally, there could be a kind of file manager that, for example, allows uploading images (or other files) that can then be referenced in the block. (imagine for example the use in an environment that has no internet connection and where we intended to use the awesome font in our articles) This "sharable" local items could be used in several blocks. 

Thanks a lot for your work.

**Comment by tangjeff0** [comment]
Can you elaborate on what you are trying to do by uploading? There is no cloud in the current version of Athens. There is only your local filesystem (and remote URLs). Therefore, the object you're trying to upload already exists in your computer.

**Comment by bserrao** [comment]
Sure, files exist on my filesystem. 
But let's say i'm making some documentation or procedure. And i want to add (attach) a screenshot (for instance). Some file i could already have on my filesystem but that i would like to be added somewhere inside the folder i've specified for Athens.
Taking the screenshot example, i can have them saved into a temp location that gets deleted or moved afterwards, i didn't want that to impact my Athens documentation.
I can do this manually... create a folder inside Athens path, move files to that folder and then link them in Athens. But it's not practical...

**Comment by TillScout** [comment]
I second the proposal to attach files, in particular by drag-and-drop into the notes. But I am not sure if I understand the exact implementation, are you @bserrao suggesting to move/copy the file into an Athens-specific location, when it is attached? Because for me, I mainly work with colleagues on files in shared folders and would rather need a link from Athens to the file where it sits in my file system, otherwise it would generate duplicates that are not shared anymore.

**Comment by bserrao** [comment]
@TillScout,  Yes, i'm suggesting that the attachment gets copied to Athens working directory. It's saved locally if you're working on a local disk, on a remote location if Athens is running somewhere in the cloud, or in a database if that's the direction Athens chooses to follow.
I think that you should be able to save the attachments in a way that it's stored in Athens just like the text you type... I don't know if i've explained myself well.

**Comment by TillScout** [comment]
yes, I understand. However I still prefer symbolic links. Often there is a reason for the file's location or it is part of a repository. E.g. when you want to create a task to update a file in your git repository:
`{{[[TODO]]}} update {{[[file: /path/to/git/Readme.md]]}}`
you do not want the file copied elsewhere, but it should remain inside the repo.

**Comment by bserrao** [comment]
Yes, but i think that's different from attaching, by attaching you're explicitly saying you want a copy of that file within your document. 
The way i see it in the example you gave above u're just linking to a remote file, so that you can work on that file from a remote location. This would mean that your Athens working folder (or part of it) is somewere in the cloud for instance.

**Comment by TillScout** [comment]
agree, those are two different approaches. One is to attach files to Athens and have them live in the Athens database or location and the other is to link to a file somewhere in the filesystem. For myself the latter version would be important.

**Comment by seekanddefine** [comment]
Seeing more feature options like [Docear](https://docear.org/) would be mind-blowing. It calls itself an academic literature management tool, but it was so much more than that. Mind-mapping your notes to your references and being able to pull up PDFs and export highlighted sections onto the mindmap to reorganize and collate ideas transformed research for me.

**Comment by seekanddefine** [comment]
Just to be clear, Athens currently supports dragging and dropping images, but not other files.
